### PR TITLE
[ng] date inputs - add granularity option

### LINKED ISSUE #1026: [ng] date-select - selection granularity
add an input to set the wanted granularity of the date select. for example sometimes we just have the need to select `january 2020` and not `january 1st 2020`

### PR DESCRIPTION
**Pull request by lucienbertin**
add `@Input() granularity` to 

- lu-date-select component
- lu-date-picker component
- lu-calendar component
- [luDateInput] directive

to allow the end user to only select a month or a year, not necessarily a date
ez

fixes #1026

-----

left to do
- [x] luDateInput should know how to parse 03/20 on granularity month, whatever the locale
- [x] default displayer for date-select should accomodate the given granularity and just display `Jan 2020` or `janvier 2020` not `Jan 1st 2020` or `1 janvier 2020`

display
_should just display "2021", not "Jan 1, 2021"_

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
jenkins auto deploy http://lucca-front.lucca.local/PR-1044

**Comment by LuccaIntegration** [comment]
jenkins auto deploy http://lucca-front.lucca.local/PR-1044

File: packages/ng/libraries/core/src/lib/date/native/native-date.adapter.ts [hunk 1]
```diff
@@ -31,14 +31,34 @@ export class LuNativeDateAdapter extends ALuDateAdapter<Date> implements ILuDate
...
 		const groups = text.split(this._regex);
 		if (groups.length !== 3 && groups.length !== 2) { return false; }
 		try {
-			const date = parseInt(groups[this._order.date], 10);
-			const month = parseInt(groups[this._order.month], 10);
-			const year = parseInt(groups[this._order.year], 10) || new Date().getFullYear();
+			const { date, month, year } = this.extract(text, granularity);
+			// const date = parseInt(groups[this._order.date], 10);
+			// const month = parseInt(groups[this._order.month], 10);
+			// const year = parseInt(groups[this._order.year], 10) || new Date().getFullYear();
```
Comment (vergerf) [inline packages/ng/libraries/core/src/lib/date/native/native-date.adapter.ts:61]: Je pense que tu as oublié ça

File: packages/ng/libraries/core/src/lib/date/native/native-date.adapter.ts [hunk 2]
```diff
@@ -67,16 +87,17 @@ export class LuNativeDateAdapter extends ALuDateAdapter<Date> implements ILuDate
...
 
-		const groups = text.split(this._regex);
-		const date = parseInt(groups[this._order.date], 10);
-		const month = parseInt(groups[this._order.month], 10);
-		const year = parseInt(groups[this._order.year], 10) || new Date().getFullYear();
+		const { date, month, year } = this.extract(text, granularity);
+		// const groups = text.split(this._regex);
+		// const date = parseInt(groups[this._order.date], 10);
+		// const month = parseInt(groups[this._order.month], 10);
+		// const year = parseInt(groups[this._order.year], 10) || new Date().getFullYear();
```
Comment (vergerf) [inline packages/ng/libraries/core/src/lib/date/native/native-date.adapter.ts:100]: idem ici

File: packages/ng/libraries/core/src/lib/date/native/native-date.adapter.ts (hunk 1 above)
Comment (lucienbertin) [reply packages/ng/libraries/core/src/lib/date/native/native-date.adapter.ts:61]: indeed

**Comment by LuccaIntegration** [comment]
jenkins auto deploy http://lucca-front.lucca.local/PR-1044

**Comment by LuccaIntegration** [comment]
jenkins auto deploy http://lucca-front.lucca.local/PR-1044

**Pull request merged**